tests/integration/targets/dns_config/tasks/main.yml

```diff
@@ -0,0 +1,161 @@
+          - result.new_state.0.uuid == actual_uuid
+
+    - name: APPEND new 'dns_servers' to existing DNS config
+      scale_computing.hypercore.dns_config:
+        overwrite: none
+        dns_servers:
+          - ips:
+              - 9.9.9.9
+              - 0.9.9.9
+            prepend: False
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result.diff.before != result.diff.after
+          - result.new_state.0.search_domains == ["example.domain0.com", "example.domain1.com", "example.domain2.com"]
+          - result.new_state.0.server_ips == ["0.0.0.0", "0.1.1.1", "1.2.3.4", "5.6.7.8", "9.9.9.9", "0.9.9.9"]
+          - result.new_state.0.uuid == actual_uuid
+
+    - name: APPEND new 'search_domains' to existing DNS config
+      scale_computing.hypercore.dns_config:
+        overwrite: none
+        search_domains:
+          - names:
+              - example.domain3.com
+            prepend: False
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result.diff.before != result.diff.after
+          - result.new_state.0.search_domains == ["example.domain0.com", "example.domain1.com", "example.domain2.com", "example.domain3.com"]
+          - result.new_state.0.server_ips == ["0.0.0.0", "0.1.1.1", "1.2.3.4", "5.6.7.8", "9.9.9.9", "0.9.9.9"]
+          - result.new_state.0.uuid == actual_uuid
+
+    - name: PREPEND 'search_domains' and APPEND 'dns_servers' existing DNS config
+      scale_computing.hypercore.dns_config:
+        overwrite: none
+        search_domains:
+          - names:
+              - example.domain-2.com
+              - example.domain-1.com
+            prepend: True
+        dns_servers:
+          # Intentionally left out "prepend" option,
+          # because it should default to "False"
+          - ips:
+              - 8.8.8.1
+              - 8.8.8.2
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result.diff.before != result.diff.after
+          - result.new_state.0.search_domains == ["example.domain-2.com", "example.domain-1.com", "example.domain0.com", "example.domain1.com", "example.domain2.com", "example.domain3.com"]
+          - result.new_state.0.server_ips == ["0.0.0.0", "0.1.1.1", "1.2.3.4", "5.6.7.8", "9.9.9.9", "0.9.9.9", "8.8.8.1", "8.8.8.2"]
+          - result.new_state.0.uuid == actual_uuid
+
+    - name: Clear all entries by using 'overwrite'
+      scale_computing.hypercore.dns_config:
+        overwrite: all
+        search_domains:
+          - names: []
+        dns_servers:
+          - ips: []
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result.diff.before != result.diff.after
+          - result.new_state.0.search_domains == []
+          - result.new_state.0.server_ips == []
+          - result.new_state.0.uuid == actual_uuid
+
+# -----------------------------------------------------------
```

Comment:
Another corner case:
```
       overwrite: all
        search_domains:
          - names:
              - example.domain1.com
              - example.domain2.com
            prepend: True
```

Here module should fail, because combo overwrite/prepend is invalid.

Do you want (need) to have different prepend for search_domains and for dns_servers? No, it is not needed - we can invoke module twice, with different prepend value.


What about a single parameter, name like `state` (`mode` as second alternative name), possible values `set` (means overwrite), `before` (means prepend), `after` (means add after)?

Since we are able to have "partial add", we will in future likely want to have also "partial remove" (no need to implement this now; but with `state=absent`

We do have something similar in module vm_disk.py - `state` can be `present` or `absent` (those two values are very common for state), in addition we have unusual `set` value - description is "With I(state=set), the VM is reconfigured to have exactly such disks as specified".


Another corner case to test: old DNS IPs are `a, b, c`. You can now try to say `prepend`:
- `d, e`
- `a, b`
- `b, a` - is desired new state `a,b,c` or `b,a,c`? Likely it is not relevant which one is selected, but it should be tested, so that behaviour does not magically change when something will be changed in code.
- `b, c`
- `c, b`
I know there is some set() to remove duplicates, but it is not tested (no unit tests).